lib/src/usage_impl_io.dart

```diff
@@ -108,10 +108,10 @@ class IOPostHandler extends PostHandler {
 JsonEncoder _jsonEncoder = JsonEncoder.withIndent('  ');
 
 class IOPersistentProperties extends PersistentProperties {
-  File _file;
-  Map _map;
+  late File _file;
+  Map? _map;
 
-  IOPersistentProperties(String name, {String documentDirPath}) : super(name) {
+  IOPersistentProperties(String name, {String? documentDirPath}) : super(name) {
     var fileName = '.${name.replaceAll(' ', '_')}';
     documentDirPath ??= userHomeDir();
     _file = File(path.join(documentDirPath, fileName));
```

Comment:
Again, with this assignment, I'm surprised that the bang is required below.

Ah OK.  Fields don't promote (a la https://dart.dev/null-safety/understanding-null-safety).  Disregard my comments above...